### ISSUE TITLE
Memory usage issue in using runIDFs() with many jobs

### ISSUE THREAD

**Issue by liq2519** [santoshphilip/eppy#210]
As I've mentioned to @santoshphilip previously, `runIDFs()` can cause memory usage issues when the number of jobs is huge, for example in a parametric analysis. This may be because every job in the list passed to `runIDFs()` is an active IDF instance that consumes considerable memory. 

I am now able to bypass this issue by writing all the .idf files and using E+ command line interface and Python `multiprocessing` module for parallel runs, but it might still be helpful to mention this potential memory usage issue in the documentation. Hopefully a similar workaround can also be added to enable large-scale E+ runs within Eppy in the future.

**Comment by santoshphilip** [comment]
@liq2519 the code has been changed to work with less memory when running many jobs. This change is in the `develop` branch. 
Can you test it with a number of jobs and let us know how it goes.
I'll close this issue once I hear back about your tests

**Comment by jamiebull1** [comment]
@liq2519 the important thing is to pass a generator to `runIDFs`, rather
than a list. The docstring wasn't clear that this was possible before.

**Comment by santoshphilip** [comment]
@liq2519 if you don't know about generators, we'll make an example file for you.

**Comment by liq2519** [comment]
Sorry wasn't able to respond earlier - just realized to turn on email notification. I'll check the documentation about generators and give it a try in the next few days.

**Comment by santoshphilip** [comment]
I have been struggling to recreate this error.

Can you try it again, but make sure that the folder `output` exists before you run.
I mean the folder `output` in the code snippet `'output_directory': 'output\\out{}'.format(i)`

**Comment by liq2519** [comment]
@santoshphilip Just noticed that I made a mistake in the weather file extension: it should be `'wtr{}.epw'.format{i}` rather than `'wtr{}.idf'.format{i}`. I believe this caused the previous error.

Now I've been able to run the toy problem as well as my own simulations of 100 idfs on 4 cores, and both of them worked well without any memory issues. Thank you very much for your help!

**Comment by santoshphilip** [comment]
I am glad that this is working. Starting up a 100 or 1000 runs and having it run with no problem is great. Very useful function.

I must say that it took me a while to get it working. It is pretty finicky about the arguments it gets and is not clear telling you what went wrong. I had to look at the test functions to figure out how to run it correctly. 

It may be worth updating the documentation to be more clear on the usage. Another thought is to update API so that it can be more forgiving. I am considering opening issues on that

**Comment by polinasali** [comment]
Hi, 
I am new to python and eppy. I am trying to run files in parallel as an example before, but get an error, what is the format structure ? Can you please help?



`from eppy import modeleditor
from eppy.modeleditor import IDF
from eppy.runner import run_functions

modeleditor.IDF.setiddname('C:\\EnergyPlusV8-9-0\\Energy+.idd')

idfsourceFolder1 = "C:\\Users\\SALI5786\\OneDrive Corp\\OneDrive - Atkins Ltd\\Desktop\\energyplus\\collection\\output_idf1\\Iteration-001"        
idfname1 = "\\2019-07-31_Change1.idf"
idfsourceFolder2 = "C:\\Users\\SALI5786\\OneDrive Corp\\OneDrive - Atkins Ltd\\Desktop\\energyplus\\collection\\output_idf1\\Iteration-002"        
idfname2 = "\\2019-07-31_Change2.idf"

epwfile = full_path

jobs = []
for i in range(1):
    jobs.append(
        [
            modeleditor.IDF(fname1.format(i), epwfile.format(i)),
            modeleditor.IDF(fname2.format(i), epwfile.format(i))
            {'output_directory': 'C:/Users/SALI5786/OneDrive Corp/OneDrive - Atkins Ltd/Desktop/energyplus'.format(i), 'ep_version': '8-9-0'}
        ]
    )
run_functions.runIDFs(jobs, 1)`

Thank you

**Comment by jamiebull1** [comment]
@polinasali could you please open this as a new issue, and also post the traceback shown when you encounter the error?

**Comment by jamiebull1** [comment]
New issue at #253
